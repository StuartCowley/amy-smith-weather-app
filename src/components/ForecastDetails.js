import React from "react";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        <h2>{formattedDate}</h2>
      </div>
      <div className="forecast-details__temperature-min">
        <span>Min temperature: </span>
        {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__temperature-max">
        <span>Max temperature: </span>
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__humidity">
        <span>Humidity: </span>
        {humidity}%
      </div>
      <div className="forecast-details__wind-speed">
        <span>Wind speed: </span>
        {wind.speed} mph
      </div>
      <div className="forecast-details__wind-direction">
        <span>Wind direction: </span>
        {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;
