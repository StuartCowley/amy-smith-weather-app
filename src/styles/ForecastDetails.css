.forecast-details {
    margin: 100px;
    display: block;
    align-items: center;
}