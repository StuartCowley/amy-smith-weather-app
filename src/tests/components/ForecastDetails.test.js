import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 111111,
    temperature: {
      min: 12,
      max: 22
    },
    humidity: 28,
    wind: {
      speed: 12,
      direction: "south"
    }
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecast={validProps} />);

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("12°C")).toHaveClass("forecast-details__temperature-min");
    expect(getByText("22°C")).toHaveClass("forecast-details__temperature-max");
    expect(getByText("28%")).toHaveClass("forecast-details__humidity");
    expect(
      getByText((_, element) => element.textContent === "Wind speed: 12 mph")
    ).toHaveClass("forecast-details__wind-speed");
    expect(getByText("south")).toHaveClass("forecast-details__wind-direction");
  });
});
