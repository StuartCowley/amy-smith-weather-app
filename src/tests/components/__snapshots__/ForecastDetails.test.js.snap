// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      <h2>
        Thu Jan 01 1970
      </h2>
    </div>
    <div
      class="forecast-details__temperature-min"
    >
      <span>
        Min temperature: 
      </span>
      12°C
    </div>
    <div
      class="forecast-details__temperature-max"
    >
      <span>
        Max temperature: 
      </span>
      22°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      <span>
        Humidity: 
      </span>
      28%
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      <span>
        Wind speed: 
      </span>
      12 mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      <span>
        Wind direction: 
      </span>
      south
    </div>
  </div>
</DocumentFragment>
`;
